### ISSUE TITLE
does giada really link against Xft, Xrender, Xcursor, Xinerama?

### ISSUE THREAD

**Issue by dvzrv** [monocasual/giada#352]
I have just cleaned the fltk package on Arch Linux from any unneeded dependencies.

After that giada can not be built without explicitely adding libxft, libxrender, libxcursor and libxinerama:

```
g++     -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -o giada src/core/giada-midiDispatcher.o src/core/giada-midiMapConf.o src/core/giada-midiEvent.o src/core/giada-audioBuffer.o src/core/giada-quantizer.o src/core/giada-conf.o src/core/giada-kernelAudio.o src/core/giada-pluginHost.o src/core/giada-pluginManager.
o src/core/giada-mixerHandler.o src/core/giada-sequencer.o src/core/giada-init.o src/core/giada-plugin.o src/core/giada-wave.o src/core/giada-waveFx.o src/core/giada-kernelMidi.o src/core/giada-graphics.o src/core/giada-patch.o src/core/giada-recorderHandler.o src/core/giada-recorder.o src/core/giada-mixer.o src/core/
giada-clock.o src/core/giada-waveManager.o src/core/giada-recManager.o src/core/channels/giada-state.o src/core/channels/giada-sampleActionRecorder.o src/core/channels/giada-midiActionRecorder.o src/core/channels/giada-waveReader.o src/core/channels/giada-midiController.o src/core/channels/giada-sampleController.o src
/core/channels/giada-samplePlayer.o src/core/channels/giada-audioReceiver.o src/core/channels/giada-midiLighter.o src/core/channels/giada-midiLearner.o src/core/channels/giada-midiSender.o src/core/channels/giada-midiReceiver.o src/core/channels/giada-channel.o src/core/channels/giada-channelManager.o src/core/model/g
iada-model.o src/core/model/giada-storage.o src/core/giada-idManager.o src/glue/giada-events.o src/glue/giada-main.o src/glue/giada-io.o src/glue/giada-storage.o src/glue/giada-channel.o src/glue/giada-plugin.o src/glue/giada-recorder.o src/glue/giada-sampleEditor.o src/glue/giada-actionEditor.o src/gui/dialogs/giada-
window.o src/gui/giada-dispatcher.o src/gui/giada-updater.o src/gui/giada-model.o src/gui/dialogs/giada-keyGrabber.o src/gui/dialogs/giada-about.o src/gui/dialogs/giada-mainWindow.o src/gui/dialogs/giada-beatsInput.o src/gui/dialogs/giada-warnings.o src/gui/dialogs/giada-bpmInput.o src/gui/dialogs/giada-channelNameInp
ut.o src/gui/dialogs/giada-config.o src/gui/dialogs/giada-devInfo.o src/gui/dialogs/giada-pluginList.o src/gui/dialogs/giada-pluginWindow.o src/gui/dialogs/giada-sampleEditor.o src/gui/dialogs/giada-pluginWindowGUI.o src/gui/dialogs/giada-pluginChooser.o src/gui/dialogs/actionEditor/giada-baseActionEditor.o src/gui/di
alogs/actionEditor/giada-sampleActionEditor.o src/gui/dialogs/actionEditor/giada-midiActionEditor.o src/gui/dialogs/browser/giada-browserBase.o src/gui/dialogs/browser/giada-browserDir.o src/gui/dialogs/browser/giada-browserLoad.o src/gui/dialogs/browser/giada-browserSave.o src/gui/dialogs/midiIO/giada-midiOutputBase.
o src/gui/dialogs/midiIO/giada-midiOutputSampleCh.o src/gui/dialogs/midiIO/giada-midiOutputMidiCh.o src/gui/dialogs/midiIO/giada-midiInputBase.o src/gui/dialogs/midiIO/giada-midiInputChannel.o src/gui/dialogs/midiIO/giada-midiInputMaster.o src/gui/elems/midiIO/giada-midiLearner.o src/gui/elems/midiIO/giada-midiLearner
Pack.o src/gui/elems/giada-browser.o src/gui/elems/giada-soundMeter.o src/gui/elems/plugin/giada-pluginBrowser.o src/gui/elems/plugin/giada-pluginParameter.o src/gui/elems/plugin/giada-pluginElement.o src/gui/elems/sampleEditor/giada-waveform.o src/gui/elems/sampleEditor/giada-waveTools.o src/gui/elems/sampleEditor/gi
ada-volumeTool.o src/gui/elems/sampleEditor/giada-boostTool.o src/gui/elems/sampleEditor/giada-panTool.o src/gui/elems/sampleEditor/giada-pitchTool.o src/gui/elems/sampleEditor/giada-rangeTool.o src/gui/elems/sampleEditor/giada-shiftTool.o src/gui/elems/actionEditor/giada-baseActionEditor.o src/gui/elems/actionEditor/
giada-baseAction.o src/gui/elems/actionEditor/giada-envelopeEditor.o src/gui/elems/actionEditor/giada-velocityEditor.o src/gui/elems/actionEditor/giada-envelopePoint.o src/gui/elems/actionEditor/giada-pianoRoll.o src/gui/elems/actionEditor/giada-noteEditor.o src/gui/elems/actionEditor/giada-pianoItem.o src/gui/elems/a
ctionEditor/giada-sampleActionEditor.o src/gui/elems/actionEditor/giada-sampleAction.o src/gui/elems/actionEditor/giada-gridTool.o src/gui/elems/mainWindow/giada-mainIO.o src/gui/elems/mainWindow/giada-mainMenu.o src/gui/elems/mainWindow/giada-mainTimer.o src/gui/elems/mainWindow/giada-mainTransport.o src/gui/elems/ma
inWindow/giada-beatMeter.o src/gui/elems/mainWindow/keyboard/giada-channelMode.o src/gui/elems/mainWindow/keyboard/giada-channelButton.o src/gui/elems/mainWindow/keyboard/giada-channelStatus.o src/gui/elems/mainWindow/keyboard/giada-keyboard.o src/gui/elems/mainWindow/keyboard/giada-column.o src/gui/elems/mainWindow/k
eyboard/giada-sampleChannel.o src/gui/elems/mainWindow/keyboard/giada-midiChannel.o src/gui/elems/mainWindow/keyboard/giada-channel.o src/gui/elems/mainWindow/keyboard/giada-sampleChannelButton.o src/gui/elems/mainWindow/keyboard/giada-midiChannelButton.o src/gui/elems/config/giada-tabMisc.o src/gui/elems/config/giada
-tabMidi.o src/gui/elems/config/giada-tabAudio.o src/gui/elems/config/giada-tabBehaviors.o src/gui/elems/config/giada-tabPlugins.o src/gui/elems/basics/giada-scroll.o src/gui/elems/basics/giada-pack.o src/gui/elems/basics/giada-group.o src/gui/elems/basics/giada-scrollPack.o src/gui/elems/basics/giada-boxtypes.o src/g
ui/elems/basics/giada-baseButton.o src/gui/elems/basics/giada-statusButton.o src/gui/elems/basics/giada-button.o src/gui/elems/basics/giada-resizerBar.o src/gui/elems/basics/giada-input.o src/gui/elems/basics/giada-liquidScroll.o src/gui/elems/basics/giada-choice.o src/gui/elems/basics/giada-dial.o src/gui/elems/basic
s/giada-box.o src/gui/elems/basics/giada-slider.o src/gui/elems/basics/giada-progress.o src/gui/elems/basics/giada-check.o src/gui/elems/basics/giada-radio.o src/utils/giada-log.o src/utils/giada-time.o src/utils/giada-math.o src/utils/giada-gui.o src/utils/giada-fs.o src/utils/giada-ver.o src/utils/giada-string.o src
/deps/rtaudio/giada-RtAudio.o    src/giada-main.o  -lsndfile -lfltk -lXext -lX11 -lXft -lXpm -lm -ljack -lasound -lpthread -ldl -lpulse-simple -lpulse -lsamplerate -lrtmidi -lfreetype -lfontconfig -lXrender -lXfixes -lXcursor -lXinerama   -lpthread
/usr/bin/ld: cannot find -lXft
/usr/bin/ld: cannot find -lXrender
/usr/bin/ld: cannot find -lXcursor
/usr/bin/ld: cannot find -lXinerama
collect2: error: ld returned 1 exit status
make: *** [Makefile:2835: giada] Error 1
```

This is slightly puzzling, because none of them are actually directly linked against in the finished package:

```
$ find-libdeps giada-0.16.3-1-x86_64.pkg.tar.zst
libsndfile.so=1-64
libfltk.so=1.3-64
libX11.so=6-64
libXpm.so=4-64
libjack.so=0-64
libasound.so=2-64
libpthread.so=0-64
libpulse-simple.so=0-64
libpulse.so=0-64
libsamplerate.so=0-64
librtmidi.so=5-64
libstdc++.so=6-64
libm.so=6-64
libgcc_s.so=1-64
libc.so=6-64
```

Are these libraries really required? The only mention of these libs that I can find in the code base are in `Makefile.{in,am}` and/or in `src/deps/juce/modules/juce_gui_basics/juce_gui_basics.cpp`/ `src/deps/juce/modules/juce_gui_basics/native/juce_linux_X11_Windowing.cpp`.

**Comment by monocasual** [comment]
Moving this to milestone 0.17.0. Will be fixed along with #153 .

**Comment by monocasual** [comment]
The [FindFLTK](https://cmake.org/cmake/help/latest/module/FindFLTK.html) CMake module should now take care of those dependencies, if really required. @dvzrv could you please try the cmake-based build and see if you can remove the unneeded libraries?

**Comment by monocasual** [comment]
Closing this as hopefully fixed by #153 . Please reopen the issue persists.
